### PR TITLE
feat: update zfs extension to v2.2.4

### PR DESCRIPTION
**Pull request by bernardgut**
see https://github.com/openzfs/zfs/releases/tag/zfs-2.2.4

### CODE REVIEW

File: Pkgfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ format: v1alpha2
 vars:
   LINUX_FIRMWARE_VERSION: "20240410" # update this when updating PKGS_VERSION in Makefile
   DRBD_DRIVER_VERSION: 9.2.8 # update this when updating PKGS_VERSION in Makefile
-  ZFS_DRIVER_VERSION: 2.2.3 # update this when updating PKGS_VERSION in Makefile
+  ZFS_DRIVER_VERSION: 2.2.4 # update this when updating PKGS_VERSION in Makefile
```
Comment (frezbo) [inline Pkgfile:8]: the checksums also needs to be updated at https://github.com/siderolabs/extensions/blob/main/storage/zfs/zfs-tools/pkg.yaml#L14, also this needs updating first https://github.com/siderolabs/pkgs/blob/main/Pkgfile#L173

Comment (bernardgut) [reply Pkgfile:8]: https://github.com/siderolabs/pkgs/pull/952

**Review by smira — APPROVED** [review]

**Pull request merged**